tests/pytype_test.py

```diff
@@ -83,34 +106,56 @@ def communicate(self):
         return self.results
 
 
+def _get_relative(filename):
+    top = filename.find('stdlib')
+    return filename[top:]
+
+
 def _get_module_name(filename):
-    """Converts a filename stdblib/m.n/module/foo to module.foo."""
-    return '.'.join(filename.split(os.path.sep)[2:]).replace(
+    """Converts a filename stdlib/m.n/module/foo to module.foo."""
+    return '.'.join(_get_relative(filename).split(os.path.sep)[2:]).replace(
         '.pyi', '').replace('.__init__', '')
 
 
 def pytype_test(args):
-    try:
-        BinaryRun(['pytd', '-h']).communicate()
-    except OSError:
-        print('Cannot run pytd. Did you install pytype?')
+    dirs = get_project_dirs(args)
+    pytype_exe = os.path.join(dirs.pytype, 'pytype')
+    pytd_exe = os.path.join(dirs.pytype, 'pytd')
+    stdlib_path = os.path.join(dirs.typeshed, 'stdlib')
+
+    if not os.path.isdir(stdlib_path):
+        print('Cannot find typeshed stdlib at %s '
+            '(specify parent dir via --typeshed_location)' % stdlib_path)
         return 0, 0
 
-    skip, parse_only = load_blacklist()
+    try:
+        BinaryRun([pytd_exe, '-h']).communicate()
+    except OSError:
+        # See if pytd is named pytd_tool instead.
+        pytd_exe = os.path.join(dirs.pytype, 'pytd_tool')
+        try:
+            BinaryRun([pytd_exe, '-h']).communicate()
+        except OSError:
+            print('Cannot run pytd. Did you install pytype?')
+            return 0, 0
```

Comment:
I think this might be a bit more readable if you did a "can_run" method that does the try/BinaryRun/except part and returns True/False, since then you can do:

```
if can_run(config.pytype, 'pytd', '-h'):
  pytd_exe = os.path.join(config.pytype, 'pytd')
elif can_run(config.pytype, 'pytd_tool', '-h'):
  pytd_exe = os.path.join(config.pytype, 'pytd_tool')
else:
  print ('Cannot run pytd. Did you install pytype?')
  return 0, 0
```

Reply:
done